File: 2022/06.ts
function findMarkerPosition(buffer: string, markerSize: number): number {
  for (let i = 0; i < buffer.length - markerSize; i++) {
    const slice = buffer.slice(i, i + markerSize);
    if (new Set(slice).size === markerSize) {
      return markerSize + i;
    }
  }
  return 0;
}

const buffer = await Deno.readTextFile("./inputs/06.txt");
console.log(findMarkerPosition(buffer, 4));
console.log(findMarkerPosition(buffer, 14));

File: 2022/06.py
def find_marker(buffer, marker_size):
    for i in range(len(buffer) - marker_size):
        slice = buffer[i : i + marker_size]
        if len(set(slice)) == marker_size:
            return marker_size + i


if __name__ == "__main__":
    buffer = open("inputs/06.txt").read()
    print(find_marker(buffer, 4))
    print(find_marker(buffer, 14))
